pkg/controller.v1alpha3/suggestion/suggestionclient/suggestionclient.go

```diff
@@ -60,8 +64,13 @@ func (g *General) SyncAssignments(
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
 
+	// Algorithm settings in suggestion will overwrite the settings in experiment.
+	filledE := e.DeepCopy()
+	appendAlgorithmSettingsFromSuggestion(filledE, instance.Spec.AlgorithmSpec)
+	experiment := g.ConvertExperiment(e)
```

Comment:
s/g.ConvertExperiment(e)/g.ConvertExperiment(filledE)

Reply:
Yeah, good find. I will update it when CI is finished. Want to see the result of CI first

Reply:
how about remove filledE, and update g.ConvertExperiment(e) to g.ConvertExperiment(e, instance.Spec.AlgorithmSpec)?

Reply:
I think code here is more readable. Or we will update the convertAlgortihmSettings. It is hard to read, I think.

Reply:
ok for me